src/MobxPromise.ts

```diff
@@ -139,10 +138,10 @@ export class MobxPromiseImpl<R>
 	@computed get result():R|undefined
 	{
 		// checking status may trigger invoke
-		if (this.isError || this.internalResult == null)
+		if (this.isError || !this.internalResult.isSet)
```

Comment:
I think this is much better, to be able to handle the actual result of invoke being `null`